python/analyzers/TreeProducerBase.py

```diff
@@ -158,13 +173,13 @@ def fillL1object(self, tree, p_name, l1obj):
     # lepton
     def bookLepton(self, tree, p_name):
         self.bookParticle(tree, p_name)
-        self.bookParticle(tree, p_name + '_jet')
+        self.bookJet(tree, p_name + '_jet')
         self.bookGeneric(tree, lepton_vars, p_name)
 
     def fillLepton(self, tree, p_name, lepton):
         self.fillParticle(tree, p_name, lepton)
-        if hasattr(lepton, 'jet'):
-            self.fillParticle(tree, p_name + '_jet', lepton.jet)
+        if (hasattr(lepton, 'jet') and lepton != lepton.jet):
```

Comment:
uhm, can't get this check. How can `lepton` be `lepton.jet` of itself?

Reply:
so the lepton.jet object is created here: [https://github.com/dehuazhu/HNL/blob/4a95b98f82efaf5fa07e83c1aaa7dba7a792e8c0/python/analyzers/JetAnalyzer.py#L164-L171](url)
It simply states that whenever there's no matched jet, the lepton is matched to himself as his own jet. We have now 2 options: 

1. We keep the code as it is and use the sanity check provided in this update
2. We change instead the JetAnalzer such that it simply doesn't assign a jet to the lepton if it doesn't has one.

Reply:
ok, right, my stuff again

Reply:
shall we resolve this conversation?